src/Workspaces/Core/Portable/FindSymbols/Declarations/DeclarationFinder_SourceDeclarations.cs

```diff
@@ -82,6 +81,31 @@ internal static partial class DeclarationFinder
                 project, name, ignoreCase, criteria, cancellationToken).ConfigureAwait(false);
         }
 
+        public static async Task<ImmutableArray<SymbolAndProjectId>> FindSourceDeclarationsWithPatternAsync(
+            Solution solution, string pattern, SymbolFilter criteria, CancellationToken cancellationToken)
```

Comment:
:bulb: I couldn't find any documentation for the semantics of `pattern`. Since (I believe) this code uses the pattern matcher, it will be important to note that the semantics of the supported pattern syntax may change over time. Here is an example of the language I'm thinking of:

> The set of symbols matched by a given `pattern` may change between releases. For example, new symbols may be matched by a pattern and/or symbols previously matched by a pattern no longer are. However, the set of symbols matched by a specific release will be consistent for a specific pattern.

Reply:
Yes, definitely. I'll add something like this.

Reply:
I added your text to the public APIs on SymbolFinder. This particular API is on an internal class, so I didn't document this one.